docs/SPEC.md

```diff
@@ -3567,6 +3573,18 @@ r0 = from(bucket:"telegraf/autogen")
 x = r0._field + "--" + r0._measurement
 ```
 
+##### truncateTimeColumn 
+
+Truncates all entries in a given time column to a specified unit.
+
+Example: 
+```
+from(bucket:"telegraph/autogen")
+    |> truncateTimeColumn(timeColumn: "_time", unit: 1s)
+```
+
+Currently, `truncateTimeColumn` only works properly if the given `timeColumn` is "_time".
```

Comment:
Yeah, I see your point. I would rephrase into:

```
Currently, `truncateTimeColumn` only works with the default time column `_time`.
```

and I would change the example to:

```
from(bucket:"telegraph/autogen")
    |> truncateTimeColumn(unit: 1s)
```

If we omit `timeColumn` from the doc, it is not a big deal, because, for now, nobody can use it.

You should also add a table with the parameters as the other functions, don't you think?

As a final note, I actually see a way to make it work with any column:

```
truncateTimeColumn = (timeColumn="_time", unit, tables=<-) => tables
        |> map(fn:(r) => ({r with __truncated_time: date.truncate(t: r._time, unit: unit)}))
        |> duplicate(column: "__truncated_time", as: timeColumn) // overwrites timeColumn
        |> drop(columns: ["__truncated_time"])
```

If the  user already has a column called `__truncated_time` that could be an issue though 😓 . What do you think?

FYI @aanthony1243 .

Reply:
`date.truncate` still takes in only `_time`

this just changes `timeColumn` from the column you want to truncate to what the resulting column is named

Reply:
Yeah, I am sorry, I meant:

```
truncateTimeColumn = (timeColumn="_time", unit, tables=<-) => tables
        |> duplicate(column: timeColumn, as: "__truncated_time")
        |> map(fn:(r) => ({r with __truncated_time: date.truncate(t: r.__truncated_time, unit: unit)}))
        |> duplicate(column: "__truncated_time", as: timeColumn) // overwrites timeColumn
        |> drop(columns: ["__truncated_time"])
```

Anyways, this is a workaround that could overwrite a column with the same name by the user. We need to support labels for this #1122 .

Only brainstorming 😄